skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt

```diff
@@ -50,13 +51,20 @@ fun isSkaffoldFile(file: VirtualFile): Boolean {
     return false
 }
 
-/**
- * Finds all Skaffold configuration YAML files in the given project.
- *
- * @param project IDE project to search Skaffold file in
- * @return List of Skaffold configuration files in the project.
- */
-fun findSkaffoldFiles(project: Project): List<VirtualFile> {
-    return FileTypeIndex.getFiles(YAMLFileType.YML, GlobalSearchScope.allScope(project))
-        .filter { isSkaffoldFile(it) }
+/** Obtains current active implementation of [SkaffoldFileService] */
+fun getSkaffoldFileService() = ServiceManager.getService(SkaffoldFileService::class.java)!!
```

Comment:
instead of a `getSkaffoldFileService` here, why not a `SkaffoldFileServce#getInstance`?

Reply:
If I understand, there are no static functions in Kotlin, so this call needs to be wrapped into companion object somehow, and I used the plain function instead. I didn't use `getInstance()` to avoid confusion. Interesting question though, @keyboardsurfer, what is your opinion? Is it better to use companion object and get the single instance of this service or use of the separate function is allowed as well?

Reply:
seems like a companion object is perfect for this use case, but I'll defer to the Kotlin experts